### ISSUE TITLE
restructing test_models.py

### ISSUE THREAD

**Issue by andrewkern** [popgensims/stdpopsim#114]
The way we have setup the generic mixin class breaks the current implementation of `models.all_models()` as that utility function gathers all the subclasses of `Model`

https://github.com/popgensims/stdpopsim/blob/77917bccfaf8ced2b39f8f2106af9fa91f9bc437/stdpopsim/models.py#L267

The basic issue is that we now have organism specific models that are subclasses of `Model` themselves, and the different demographic models sit as subclasses underneath there in the inheritance. 

I'm can rewrite the `all_models()` function to deal with this but it raises another issue-- do we want to have a list of the species included in stdpopsim somewhere? just something so that we could get a list using `stdpopsim.species.all_species()`?

**Comment by jeromekelleher** [comment]
Not sure what the best thing to do is here. Will take a look.

**Comment by jeromekelleher** [comment]
No relevant  after #110.
